static/app/components/events/featureFlags/featureFlagDrawer.tsx

```diff
@@ -74,6 +74,22 @@ interface FlagDrawerProps {
   hydratedFlags: KeyValueDataContentProps[];
   initialSort: FlagSort;
   project: Project;
+  focusControl?: FlagControlOptions;
+}
+
+function useFocusControl(initialFocusControl?: FlagControlOptions) {
+  const [focusControl, setFocusControl] = useState(initialFocusControl);
+  // If the focused control element is blurred, unset the state to remove styles
+  // This will allow us to simulate :focus-visible on the button elements.
+  const getFocusProps = useCallback(
+    (option: FlagControlOptions) => {
+      return option === focusControl
+        ? {autoFocus: true, onBlur: () => setFocusControl(undefined)}
+        : {};
+    },
+    [focusControl]
+  );
+  return {getFocusProps};
```

Comment:
this is the same function as in `breadcrumbsDrawer.tsx`:
https://github.com/getsentry/sentry/blob/f98fe827f109f9946aaeeca3473f6436c4f58fc1/static/app/components/events/breadcrumbs/breadcrumbsDrawer.tsx#L52-L65

i _could_ rewrite the function to be more generic so that it can be imported by both files. don't know if anyone has a super strong opinion about this

Reply:
I think a generic function would be better, but also no strong opinion on this